### PR TITLE
[NEW] Support autocompletion in new composer

### PR DESCRIPTION
**Pull request by cardoso**
@RocketChat/ios

I had to refactor AvatarView to be programmatic so it can work with UserHintCell<AvatarView>

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/MessagesViewControllerComposerDelegate.swift [hunk 1]
```diff
@@ -24,27 +25,102 @@ extension MessagesViewController: ComposerViewExpandedDelegate {
...
+            commands.forEach {
+                viewModel.hints.append($0.command)
+            }
+        } else if prefix == ":" {
+            let emojis = EmojiSearcher.standard.search(shortname: word.lowercased(), custom: CustomEmoji.emojis())
+
+            emojis.forEach {
+                viewModel.hints.append($0.suggestion)
+            }
+        }
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/MessagesViewControllerComposerDelegate.swift]: What do you think about moving this method to the view model as well? Maybe we can have one view model just for the hints? 🤔

File: Rocket.Chat/Views/Avatar/AvatarView.swift [hunk 2]
```diff
@@ -179,13 +210,12 @@ final class AvatarView: UIView {
...
 }
 
 // MARK: Themeable
 
 extension AvatarView {
     override func applyTheme() {
-        labelInitials?.textColor = .white
+        labelInitials.textColor = .white
     }
 }
```
Comment (rafaelks) [inline Rocket.Chat/Views/Avatar/AvatarView.swift]: 👌

File: Rocket.Chat/Controllers/Chat/MessagesViewControllerComposerDelegate.swift (hunk 1 above)
Comment (cardoso) [reply Rocket.Chat/Controllers/Chat/MessagesViewControllerComposerDelegate.swift]: I saw that you guys are doing this pattern, so I did the same. I created a MessageComposerViewModel and put this logic there :+1:

File: Rocket.Chat/Views/Avatar/AvatarView.swift (hunk 2 above)
Comment (rafaelks) [reply Rocket.Chat/Views/Avatar/AvatarView.swift]: I believe you need to call super here... don't you?

File: Rocket.Chat/Controllers/Chat/MessagesViewControllerComposerDelegate.swift (hunk 1 above)
Comment (rafaelks) [reply Rocket.Chat/Controllers/Chat/MessagesViewControllerComposerDelegate.swift]: 👍

File: Rocket.Chat/Views/Avatar/AvatarView.swift (hunk 2 above)
Comment (cardoso) [reply Rocket.Chat/Views/Avatar/AvatarView.swift]: Not needed in this particular case I think, but I can do it for consistency.

**Pull request merged**